cms/api.py

```diff
@@ -0,0 +1,174 @@
+        )
+    }
+    return (
+        sorted(
+            valid_program_pages + valid_course_pages + valid_external_course_pages,
+            # ProgramPages with the same next run date as a CoursePage should be sorted first
+            key=lambda page: (
+                page_run_dates[page],
+                page.is_course_page or page.is_external_course_page,
+                page.title,
+            ),
+        ),
+        sorted(
+            valid_program_pages, key=lambda page: (page_run_dates[page], page.title)
+        ),
+        sorted(
+            valid_course_pages + valid_external_course_pages,
+            key=lambda page: (page_run_dates[page], page.title),
+        ),
+    )
+
+
+def get_home_page():
+    """
+    Returns an instance of the home page (all of our Wagtail pages are expected to be descendants of this home page)
+
+    Returns:
+        Page: The home page object
+    """
+    return Page.objects.get(
+        content_type=ContentType.objects.get_for_model(cms_models.HomePage)
+    )
+
+
+def ensure_home_page_and_site():
+    """
+    Ensures that Wagtail is configured with a home page of the right type, and that
+    the home page is configured as the default site.
+    """
+    site = Site.objects.filter(is_default_site=True).first()
+    valid_home_page = Page.objects.filter(
+        content_type=ContentType.objects.get_for_model(cms_models.HomePage)
+    ).first()
+    root = Page.objects.get(depth=1)
+    if valid_home_page is None:
+        valid_home_page = cms_models.HomePage(**DEFAULT_HOMEPAGE_PROPS)
+        root.add_child(instance=valid_home_page)
+        valid_home_page.refresh_from_db()
+    if site is None:
+        Site.objects.create(
+            is_default_site=True, root_page=valid_home_page, **DEFAULT_SITE_PROPS
+        )
+    elif site.root_page is None or site.root_page != valid_home_page:
+        site.root_page = valid_home_page
+        site.save()
+        log.info("Updated site: %s", site)
+    wagtail_default_home_page = Page.objects.filter(
+        depth=2, content_type=ContentType.objects.get_for_model(Page)
+    ).first()
+    if wagtail_default_home_page is not None:
+        wagtail_default_home_page.delete()
+
+
+def ensure_catalog_page():
+    """
+    Ensures that a catalog page with the correct slug exists. If this page doesn't
+    exist with the correct slug, the course catalog cannot be accessed.
+    """
+    catalog_page = Page.objects.filter(
+        content_type=ContentType.objects.get_for_model(cms_models.CatalogPage)
+    ).first()
+    if catalog_page is not None and catalog_page.slug != "catalog":
+        catalog_page.delete()
+        catalog_page = None
+    if catalog_page is None:
+        catalog_page = cms_models.CatalogPage(title="Catalog")
+        home_page = get_home_page()
+        home_page.add_child(instance=catalog_page)
+        catalog_page.refresh_from_db()
+
+
+def ensure_index_pages():
```

Comment:
Can we please ensure the `CertificateIndexPage` is also created here?

Reply:
Great catch. Updated